src/client/transaction/transaction.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package transaction;
+option go_package = "github.com/pachyderm/pachyderm/src/client/transaction";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+
+import "gogoproto/gogo.proto";
+
+import "client/pfs/pfs.proto";
+
+message DeleteAllRequest {
+}
+
+message TransactionRequest {
+  oneof request {
```

Comment:
Up until this point we've opted to not use `oneof` and just do an "implicit oneof" by having a message with multiple fields in which only one should ever be set but that isn't enforced by the type checker. If I recall correctly we made that decision because the code required to interact with `oneof`s in go was pretty cumbersome and didn't seem worth the extra safety. We might want to rethink that but I think it'd be good for this code to be unified with what we already have and the path of least resistance would be to change this code rather than all the other stuff.

Reply:
I found the code for dealing with `oneof` to be not too cumbersome, at least vs trying to enforce such an invariant in code.  That is, when we add more supported requests and responses to the message, I would expect non-`oneof` code to need to scale O(n^2), while `oneof` code should scale O(n).  I could be wrong, though, if you have some prior art you can point me to.  I'm not opposed to changing this, but I don't want the code to become unmaintainable.